src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -1463,7 +1430,11 @@ private IExpressionStatement CreateBoundExpressionStatementOperation(BoundExpres
             SyntaxNode syntax = boundExpressionStatement.Syntax;
             ITypeSymbol type = null;
             Optional<object> constantValue = default(Optional<object>);
-            bool isImplicit = boundExpressionStatement.WasCompilerGenerated;
+
+            // lambda body can point to expression directly and binder can insert exression statement there. and end up statement pointing to
+            // expression syntax node since there is no statement syntax node to point to. this will mark such one as implicit since it doesn't
+            // actually exist in code
+            bool isImplicit = boundExpressionStatement.WasCompilerGenerated || !boundExpressionStatement.Syntax.IsKind(SyntaxKind.ExpressionStatement);
```

Comment:
>IsKind(SyntaxKind.ExpressionStatement) [](start = 112, length = 38)

This condition is probably too strict, we might want to simply check if boundExpressionStatement.Expression has the same syntax.

Reply:
sure.